src/git-repository-async.js

```diff
@@ -0,0 +1,191 @@
+'use babel'
+
+const Git = require('nodegit')
+const path = require('path')
+const {Emitter, Disposable, CompositeDisposable} = require('event-kit')
+
+// Temporary requires
+// ==================
+// GitUtils is temporarily used for ::relativize only, because I don't want
+// to port it just yet. TODO: remove
+const GitUtils = require('git-utils')
+// Just using this for _.isEqual and _.object, we should impl our own here
+const _ = require('underscore-plus')
+
+module.exports = class GitRepositoryAsync {
+  static open (path) {
+    // QUESTION: Should this wrap Git.Repository and reject with a nicer message?
+    return new GitRepositoryAsync(path)
+  }
+
+  constructor (path) {
+    this.repo = null
+    this.emitter = new Emitter()
+    this.subscriptions = new CompositeDisposable()
+    this.pathStatusCache = {}
+    this._gitUtilsRepo = GitUtils.open(path) // TODO remove after porting ::relativize
+    this.repoPromise = Git.Repository.open(path)
+  }
+
+  destroy () {
+    this.subscriptions.dispose()
+  }
+
+  getPath () {
+    return this.repoPromise.then((repo) => {
+      return Promise.resolve(repo.path().replace(/\/$/, ''))
+    })
+  }
+
+  isPathIgnored (_path) {
+    return this.repoPromise.then((repo) => {
+      return Promise.resolve(Git.Ignore.pathIsIgnored(repo, _path))
+    })
+  }
+
+  isPathModified (_path) {
+    return this._filterStatusesByPath(_path).then(function (statuses) {
+      var ret = statuses.filter((status) => {
```

Comment:
V8 supports the new ES6 `Array.prototype.every` method, which would be clearer and save you allocation of a new array.